ApiReview/API.Polly.Core/NoDocs/Polly/ResilienceContext.cs

```diff
@@ -0,0 +1,25 @@
+// Assembly 'Polly.Core'
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using Polly.Strategy;
+using Polly.Utils;
+
+namespace Polly;
+
+public sealed class ResilienceContext
+{
+    public CancellationToken CancellationToken { get; set; }
+    public bool IsSynchronous { get; }
+    public Type ResultType { get; }
+    public bool IsVoid { get; }
+    public bool ContinueOnCapturedContext { get; set; }
+    public ResilienceProperties Properties { get; }
+    public IReadOnlyCollection<ReportedResilienceEvent> ResilienceEvents { get; }
+    public static ResilienceContext Get();
```

Comment:
Do Get/Return need to be public, or is that used only by the infra?

If Get needs to be public, would it make sense to change the signature to:

```
public static ResilienceContext Get(CancellationToken token, bool continueOnCapturedContext);
```

This would immediately set the two associated properties as part of getting the context.

Reply:
Consider introducing analyzers that checks resilience context is returned to the pool.

Reply:
Make the setter for cancellation token internal for now.